### DISCUSSION TITLE
How to set up a plane wave source with any polarization angle

### DISCUSSION THREAD

**Discussion by Superwbb1999** [NanoComp/meep#2674] [Q&A]
How to set up a planar wave source with any polarization angle? Is it handled by multiplying the corresponding components such as Ex or Ey by the trigonometric function of the polarization angle?

**Comment by smartalecH** [comment]
See this tutorial: https://meep.readthedocs.io/en/latest/Python_Tutorials/Eigenmode_Source/#planewaves-in-homogeneous-media

**Comment by dalarev** [comment]
... and see [this](https://github.com/dalarev/MEEP_scripts/blob/main/test_offnormal_source.ipynb) script if you need an additional worked example in 3D.
